dnf/transaction_sr.py

```diff
@@ -220,25 +219,31 @@ def __init__(
         self._nevra_reason_cache = {}
         self._warnings = []
 
+    def load_from_file(self, fn):
```

Comment:
I'm not sure whether it's a better API to have `load_from_file()` and `load_from_dict()` methods aor whether it would be better to have an init like `__init__(self, base, data=None, filename=None, ...)` and then you'd specify one of [`data`, `filename`] when instantiating the class? I know there's no RAII in Python :slightly_smiling_face: but the class now requires you to call a load method and if you don't, you'll get a rather unspecific crash (though it's a programmer error).